sdk/communication/Azure.Communication.Common/tests/CommunicationIdentifierSerializerTest.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using NUnit.Framework;
+
+namespace Azure.Communication
+{
+    public class CommunicationIdentifierSerializerTest
+    {
+        [Test]
+        public void MissingProperty_DeserializerThrows()
+        {
+            CommunicationIdentifierModel[] modelsWithMissingMandatoryProperty = new[]
+            {
+                new CommunicationIdentifierModel(CommunicationIdentifierKind.Unknown), // Missing Id
+                new CommunicationIdentifierModel(CommunicationIdentifierKind.CommunicationUser), // Missing Id
+                new CommunicationIdentifierModel(CommunicationIdentifierKind.CallingApplication), // Missing Id
+                new CommunicationIdentifierModel(CommunicationIdentifierKind.PhoneNumber) { Id = "some id" }, // Missing PhoneNumber
```

Comment:
Question, in this test case, it is expected to throw an exception due to missing phone number? From the swagger phone number is not requried.

Reply:
Yes, we expect to throw. Unfortunately swagger syntax is not powerful enough to explain all the valid combinations.